mantidproject/mantid
# Add Save Settings to File Button

**Description of work.**
This PR adds a button to the settings to save the settings to a file and another to load the settings back in.

**Report to:** SANS group

**To test:**

Open workbench and open settings (`File`->`Settings`)
Change some settings and click `Save to File`. This should create a file which contains all the settings in their current state.
Change some more settings
Click `Load from File` and load in the saved file.
This should put all the settings back into the state they were when they were saved.

Fixes #28606 

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
